### ISSUE TITLE
Model Training not embedding class names into model

### ISSUE THREAD

**Issue by thomasf1** [ultralytics/ultralytics#6905]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

When training a model with 

```python
from ultralytics import YOLO

# Load a model
model = YOLO('yolov8x-seg.pt')  # load a pretrained model (recommended for training)

# Train the model
model.train(data='P5-fin-pers.yaml', epochs=100, imgsz=640)
```

and P5-fin-pers.yaml being
```yaml
categories:
- person
nc: 1
train: P5-fin/train-pers
val: P5-fin/val-pers
```

I get the following result after training. Why isn´t the class name (person) used?

```python
print('classes', model.names)
```
classes {0: '**class_0**'}

**Comment by glenn-jocher** [comment]
@thomasf1 hello! When training a YOLOv8 model, class names are generally derived from the dataset used. If you're seeing default names like 'class_0' instead of 'person', this usually indicates that the dataset's class labels were not properly communicated to the model during the training process. 

To ensure class names are embedded correctly, double-check your dataset configuration file. It should reference a `.names` file containing your class labels, or otherwise correctly specify the class labels directly within the YAML. The training process should then automatically detect and use these class labels.

The process of resolving this might include verifying the structure and content of your dataset, ensuring that the paths to your label files are correct, and that class names are specified in a format that the training script expects. For further specifics, please refer to the Ultralytics Docs. If everything seems correct on your end and the problem persists, feel free to open an issue with detailed information and we'll be glad to assist! 🌟

**Comment by thomasf1** [comment]
Thanks. Figured it out, the software creating the COCO YAML was using "categories" instead of "names".

**Comment by glenn-jocher** [comment]
Great to hear that you resolved the issue, @thomasf1! Yes, using "names" rather than "categories" in your COCO YAML is the correct way to ensure class names are embedded in the model. If you have any more questions or need further assistance, feel free to ask. Happy training! 🚀
